parlai/core/torch_generator_agent.py

```diff
@@ -603,9 +606,12 @@ def eval_step(self, batch):
         """
         Evaluate a single batch of examples.
         """
-        if batch.text_vec is None:
+        if batch.text_vec is None and batch.image is None:
```

Comment:
So we've got this in a bunch of places now. I'm wondering if life would be easier if batch world didn't give the agents padding examples, so that `is_valid` and checks like this could be completely removed. Opinions from the team?

Reply:
at the very least, can we please put a `batchsize` attr in the batch? or maybe just a `batch.valid` attr that marks whether there are any valid examples in the batch? since that is a way more clear check than just checking if text or image exists